### PR TITLE
Remove references to the V1 Android embedding

### PR DESCRIPTION
**Pull request by blasten**
Issue: https://github.com/flutter/flutter/issues/86578

Remove references to the V1 Android embedding

### CODE REVIEW

File: packages/android_intent/example/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -10,14 +10,6 @@
     android:icon="@mipmap/ic_launcher"
     android:label="android_intent_example"
     android:name="io.flutter.app.FlutterApplication">
```
Comment (bparrishMines) [inline packages/android_intent/example/android/app/src/main/AndroidManifest.xml:12]: nit: This line was removed for the rest of the plugins. Does it need to be removed here too?

**Comment by stuartmorgan** [comment]
Could you add an issue link with more context so that someone looking back at this change understands why this was done? (PRs with almost no context have been a significant problem for me as I've tried to investigate why various things in this repository are the way they are).

In this case, I thought in our previous discussion this was a multi-phase process where removing tests that we properly support the V1 embedding were going to remain for as long as we actually do support it, but it seems like those are being removed here.

**Comment by stuartmorgan** [comment]
Also, you've changed all of the deprecated plugins, which we were not planning on republishing short of a critical (e.g., security) fix. Is it actually important that we update those?

File: packages/android_intent/example/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (blasten) [reply packages/android_intent/example/android/app/src/main/AndroidManifest.xml]: fixed. I also saw additional ones. thanks

**Comment by blasten** [comment]
> Could you add an issue link with more context so that someone looking back at this change understands why this was done?

Done

> . Is it actually important that we update those?

Ideally. Although, maybe not very urgent if their checks aren't running on CI

**Comment by stuartmorgan** [comment]
> > . Is it actually important that we update those?
> 
> Ideally. Although, maybe not very urgent if their checks aren't running on CI

They are running on CI. If we need it for CI, but it's not critical for end users of the plugins, then we should update them but use `NEXT` instead of bumping the versions so that they don't publish new versions.

**Comment by blasten** [comment]
> we should update them but use NEXT instead of bumping the versions so that they don't publish new versions.

Reverted the change.

**Comment by ditman** [comment]
> > we should update them but use NEXT instead of bumping the versions so that they don't publish new versions.
> 
> Reverted the change.

I think that we still should still publish the deprecated plugins, just to ensure that we don't have lingering documentation about the V1 embedding in the repo (so if the change touches the README.md of a plugin, it should be publipshed). (Not sure how "dangerous" is to have old school documentation lying around for some extra weeks/months on those plugins, though.)

Also @blasten, the latest change modified 3 plugins of which *none* are deprecated (I think?). Here's the [list of deprecated plugins](https://github.com/flutter/plugins/blob/master/README.md#deprecated) (or are we talking about a different deprecation?)

**Comment by blasten** [comment]
ah thanks. I just reverted all of them.

**Review by stuartmorgan — COMMENTED** [review]
> I think that we still should still publish the deprecated plugins, just to ensure that we don't have lingering documentation about the V1 embedding in the repo

You mean on pub.dev? It's not in the repo whether we publish it or not.

The plugins are deprecated on pub.dev, so someone reading the instructions about integrating the plugin, on pub.dev, is already ignoring our recommendations. I'm not sure what problem we are solving by having updated instructions on starting to use plugins that we are telling people not to start using...

File: packages/android_alarm_manager/README.md [hunk 2]
```diff
@@ -82,41 +82,6 @@ alarm manager plugin itself, it may be necessary to inform the background servic
 to initialize plugins depending on which Flutter Android embedding the application is
 using.
```
[the remark this answers is not in the document]
Comment (blasten) [reply packages/android_alarm_manager/README.md]: removed the API. thanks

File: packages/google_sign_in/google_sign_in/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 * Add iOS unit and UI integration test targets.
 * Add iOS unit test module map.
 * Exclude arm64 simulators in example app.
+* Remove references to the Android V1 embedding.
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/CHANGELOG.md:6]: You need to change the version, not add to an already-published version's release notes.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 * Add iOS unit and UI integration test targets.
 * Exclude arm64 simulators in example app.
+* Remove references to the Android V1 embedding.
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:5]: This plugin isn't deprecated, so you should covert the NEXT to a version bump at this point.

File: packages/path_provider/path_provider/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
 
 * Add iOS unit test target.
+* Remove references to the Android V1 embedding.
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider/CHANGELOG.md:4]: The NEXT should become a version bump, as above.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/google_sign_in/google_sign_in/CHANGELOG.md (hunk 3 above)
Comment (blasten) [reply packages/google_sign_in/google_sign_in/CHANGELOG.md]: sorry. fixed.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md (hunk 4 above)
Comment (blasten) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:5]: done

File: packages/path_provider/path_provider/CHANGELOG.md (hunk 5 above)
Comment (blasten) [reply packages/path_provider/path_provider/CHANGELOG.md:4]: done

**Comment by blasten** [comment]
PTAL @stuartmorgan

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**